Initial AMM rule support

justinethier opened the pull request.

BrianSipos reviewed (commented):
Some design questions in the objects, can be discussed outside of github.

justinethier commented:
@BrianSipos This latest set of changes addresses your comments. I also added the `enabled` field to the "listing" EDD's in the ADM. It seemed appropriate to have a way to report on rule status as well.

BrianSipos reviewed (approved):
Thanks for taking feedback, there will probably be some additional tweaking of object fields when actually used by the exec worker.
